### PR TITLE
PIM-6489: fix the sort of attributes in attribute groups

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

Fixes the sort of attributes in attribute groups.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | Y
| Added integration tests           | N
| Changelog updated                 | Y
| Review and 2 GTM                  | Todo

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js [hunk 1]
```diff
@@ -79,19 +79,22 @@ define(
                 FetcherRegistry.getFetcher('attribute')
                     .fetchByIdentifiers(this.getFormData().attributes, {rights: 0})
                     .then(function (attributes) {
-                        var attributes = _.map(attributes, function (attribute) {
-                            //This update the sort order if the attribute is new on the collection
-                            var sortOrder = this.getFormData().attributes_sort_order[attribute.code] ?
-                                this.getFormData().attributes_sort_order[attribute.code] :
-                                _.keys(this.getFormData().attributes_sort_order) + 1;
+                        attributes = _.map(attributes, function (attribute) {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: could you please .map of es6?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js [hunk 2]
```diff
@@ -79,19 +79,22 @@ define(
...
+                            let sortOrder = this.getFormData().attributes_sort_order[attribute.code];
+
+                            //This updates the sort order if the attribute is new on the collection
+                            if (undefined === sortOrder) {
+                                sortOrder = _.keys(this.getFormData().attributes_sort_order).length + 1;
+                            }
 
                             return _.extend(
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: you can do the same with {...attribute, {sort_order: sortOrder}}

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js (hunk 1 above)
Comment (juliensnz) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: no const/var?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js [hunk 3]
```diff
@@ -79,19 +79,22 @@ define(
...
-                            //This update the sort order if the attribute is new on the collection
-                            var sortOrder = this.getFormData().attributes_sort_order[attribute.code] ?
-                                this.getFormData().attributes_sort_order[attribute.code] :
-                                _.keys(this.getFormData().attributes_sort_order) + 1;
+                        attributes = _.map(attributes, function (attribute) {
+                            let sortOrder = this.getFormData().attributes_sort_order[attribute.code];
+
+                            //This updates the sort order if the attribute is new on the collection
+                            if (undefined === sortOrder) {
+                                sortOrder = _.keys(this.getFormData().attributes_sort_order).length + 1;
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: object.keys :)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js [hunk 4]
```diff
@@ -79,19 +79,22 @@ define(
...
 
                             return _.extend(
                                 {},
                                 attribute,
                                 {sort_order: sortOrder}
                             );
                         }.bind(this));
-                        var attributes = _.sortBy(attributes, 'sort_order');
+
+                        attributes = _.sortBy(attributes, 'sort_order');
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: there is a new .sort method in es6 the code will look like this:
```
attributes.sort((first, second) => first.sort_order - second.sort_order);
```

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js (hunk 1 above)
Comment (grena) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: I'll use the map of es6, and no, the attributes is the argument of the function

Comment (SamirBoulil) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute-group/form/tab/attribute.js]: it's also the variable it is assigned to: `attributes = _.map(attributes` you should use var or const no ?

**Pull request merged**